pi4j-core/src/main/java/com/pi4j/io/spi/Spi.java

```diff
@@ -366,4 +366,59 @@ default int transfer(ByteBuffer buffer, int offset, int length) {
         // return actual number of bytes read
         return length;
     }
+
+    // ------------------------------------------------------------------------------------
+    //  Register Read
+    // ------------------------------------------------------------------------------------
+    /**
+     * This function writes multiple bytes to the SPI device and then reads bytes from the device
+     * Write data is taken from the 'buffer' byte array, data
+     * read back from the SPI device is then copied to the 'read' byte array
+     *
+     * @param buffer the array of bytes to write to the SPI device
+     * @param writeLength
+     *               Number bytes written from buffer to the SPI
+     * @param read   Buffer to contain read data
+     * @param readlength the number of bytes to read (read &amp; read))
+     * @return Returns 0 if OK, otherwise PI_BAD_HANDLE, PI_BAD_SPI_COUNT, or PI_SPI_XFER_FAILED.
+     * @see "http://abyz.me.uk/rpi/pigpio/cif.html#spiWrite"
+     */
+    default int registerRead(byte[] buffer, int writeLength,  byte[] read,  int readlength) {
```

Comment:
I guess there is already a method with similar signature `transfer` (I am using it now with ffm, but for different purpose). May be we should stick with it?

https://github.com/Pi4J/pi4j/blob/develop/pi4j-core/src/main/java/com/pi4j/io/spi/Spi.java#L128